File: test_files/java/SuggestionDocumentation.java
public class SuggestionDocumentation {
    static String normalizeName(String rawName) {
        String trimmed = rawName.trim();
        String[] parts = trimmed.split("\\s+");
        StringBuilder builder = new StringBuilder();
        for (int i = 0; i < parts.length; i += 1) {
            String part = parts[i];
            String normalized = part.substring(0, 1).toUpperCase() + part.substring(1).toLowerCase();
            if (i > 0) {
                builder.append(" ");
            }
            builder.append(normalized);
        }
        return builder.toString();
    }

    static String wrapPreview(String value) {
        return "[" + value + "]";
    }

    static String docPreview() {
        String user = normalizeName("  rina patel ");
        String badge = "eng:" + user;
        // DOC-CURSOR: trigger autocomplete after "(" and document ghost text.
        return wrapPreview(
    }
}

File: test_files/typescript/suggestion_documentation.ts
export function normalizeName(rawName: string): string {
  return rawName.trim().split(/\s+/).map((part) => {
    return part[0].toUpperCase() + part.slice(1).toLowerCase();
  }).join(" ");
}

export function wrapPreview(value: string): string {
  return `[${value}]`;
}

export function docPreview(): string {
  const user = normalizeName("  rina patel ");
  const badge = `eng:${user}`;
  // DOC-CURSOR: trigger autocomplete after "(" and document ghost text.
  return wrapPreview(
}
